### PR TITLE
Support `Column<Array<T>>`

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Requires https://github.com/heavyai/heavydb-internal/pull/7049

Add bindings for the following functions:
- `Column<Array<T>>::getitem`
- `Column<Array<T>>::setItem`
- `Column<Array<T>>::concatItem`
- `Column<Array<T>>::isNull`
- `Column<Array<T>>::setNull`
- `Column<Array<T>>::setNull`
- `TableFunctionManager::set_output_array_values_total_number`

### CODE REVIEW

File: rbc/heavydb/column_array.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+        """
+        Set the array at ``index`` to null.
+
+        .. note::
+            Only available on ``CPU``
+        """
+
+    def __len__(self) -> int:
+        """
+        """
```
Comment (pearu) [inline rbc/heavydb/column_array.py:92]:
```suggestion
        """
        Return the length of the array.
        
        ..  note::
             Only available on ``CPU``
        """
```

File: rbc/heavydb/column_array.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+    the input and output arguments of UDTFs.
+    """
+
+
+class ColumnArrayType(nb_types.Type):
+    """Numba type class for HeavyDB buffer structures.
+    """
+
+    def __init__(self, dtype):
+        array_T = self.__typesystem_type__._params.get(_COLUMN_PARAM_NAME)
```
Comment (pearu) [inline rbc/heavydb/column_array.py:134]: IIUC, `self` is `numba.core.types.Type` derived instance. Where is defined its `__typesystem_type__` attribute? In numba?

**Review by pearu — APPROVED** [review]
LGTM! Nice work, @guilhermeleobas!

I have one question though, see below.

File: rbc/heavydb/column_array.py (hunk 2 above)
Comment (guilhermeleobas) [reply rbc/heavydb/column_array.py:138]: Here:
https://github.com/xnd-project/rbc/blob/c576db2714c950b31bf799590d91bd42df768b57/rbc/structure_type.py#L93

**Pull request merged**